### PR TITLE
[quant][fx] Add node name as prefix to observer module name

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44846 [quant] Add quant APIs to save/load observer state_dict
* #44773 [quant][qat] Ensure fake_quant and observer can be disabled on scriptmodule
* **#44765 [quant][fx] Add node name as prefix to observer module name**

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23741355](https://our.internmc.facebook.com/intern/diff/D23741355)

### CODE REVIEW

File: torch/quantization/fx/quantize.py [hunk 1]
```diff
@@ -274,6 +275,7 @@ def is_observed(input_arg):
                 env[node.name] = observed_graph.node_copy(node, load_arg)
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:275]: looks like this creates a new name, can you move L240 after this line and use:
```
self.activation_post_process_map[env[node.name]] = observer
```

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py:275]: similar for the next case

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py:275]: sorry, not this one

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py:275]: the idea is the above, I'm not sure which line of code to change, please take a look.

Comment (supriyar) [reply torch/quantization/fx/quantize.py:275]: @jerryzh168 why do we need to move L240 after this? This else clause doesn't insert observers.

Comment (supriyar) [reply torch/quantization/fx/quantize.py:275]: The actual issue was somewhere else, in convert we explicitly check if module name startswith `activation_post_process`, so had to update that.

File: torch/quantization/fx/quantize.py [hunk 2]
```diff
@@ -463,7 +465,7 @@ def is_quantized(node):
 
             # handle activation post process calls
             if node.op == 'call_module':
-                if node.target.split('.')[-1].startswith('activation_post_process_'):
+                if '_activation_post_process_' in node.target.split('.')[-1]:
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:468]: maybe put this in a function?
also i'm not sure whether it's a good idea to check the name actually after a second thought, can we check the type instead?

**Review by jerryzh168 — APPROVED** [review]
please move the checks to a function, optional: change it to check the type of the modules[node.target]

**Pull request closed without merging**